core/src/main/java/org/bitcoinj/wallet/KeyChainGroup.java

```diff
@@ -437,13 +437,13 @@ public Address freshAddress(KeyChain.KeyPurpose purpose) {
         if (chain.isMarried()) {
             Script outputScript = chain.freshOutputScript(purpose);
             checkState(ScriptPattern.isP2SH(outputScript)); // Only handle P2SH for now
-            Address freshAddress = LegacyAddress.fromScriptHash(params,
+            Address freshAddress = LegacyAddress.fromScriptHash(params.network(),
                     ScriptPattern.extractHashFromP2SH(outputScript));
             maybeLookaheadScripts();
             currentAddresses.put(purpose, freshAddress);
             return freshAddress;
         } else if (outputScriptType == ScriptType.P2PKH || outputScriptType == ScriptType.P2WPKH) {
-            return Address.fromKey(params, freshKey(purpose), outputScriptType);
```

Comment:
There is a deprecated API usage migrationl snuck in here. I'll prepare a commit that migrates all of them.

Reply:
OK, thanks!

Reply:
See https://github.com/bitcoinj/bitcoinj/pull/2616